src/events.jl

```diff
@@ -60,26 +74,60 @@ function readevents(path; T = Float64)
 
             # Check if the HDU is a table
             if isa(hdu, TableHDU)
-                # Get column names using the correct FITSIO method
                 colnames = FITSIO.colnames(hdu)
 
-                if "TIME" in colnames
+                # Read TIME and ENERGY data if columns exist and vectors are empty
+                if isempty(times) && ("TIME" in colnames)
                     times = convert(Vector{T}, read(hdu, "TIME"))
                 end
-                if "ENERGY" in colnames
+                if isempty(energies) && ("ENERGY" in colnames)
                     energies = convert(Vector{T}, read(hdu, "ENERGY"))
                 end
+
+                # If we found both time and energy data, we can return
+                if !isempty(times) && !isempty(energies)
```

Comment:
Note that most event lists do _not_ have an "energy" column, so this will keep going through the different extension needlessly. 
When you find the time column, you will also have found all the information you can find about single events (energy, PI, etc.).